include/utransport/UPriority.h

```diff
@@ -0,0 +1,154 @@
+#ifndef _UPRIORITY_H_
+#define _UPRIORITY_H_
+
+#include <iostream>
+#include <optional>
+
+namespace uprotocol 
+{
+    namespace utransport
+    {
+        namespace datamodel
+        {
+            enum UPriority {
+                LOW =  0,                   // Low Priority. No bandwidth assurance such as File Transfer.
+                STANDARD = 1,               // Standard, undifferentiated application such as General (unclassified).
+                OPERATIONS =  2,            // Operations, Administration, and Management such as Streamer messages (sub, connect, etc…)
+                MULTIMEDIA_STREAMING = 3,   // Multimedia streaming such as Video Streaming
+                REALTIME_INTERACTIVE = 4,   // Real-time interactive such as High priority (rpc events)
+                SIGNALING =  5,             // Signaling such as Important
+                NETWORK_CONTROL =  6,       // Network control such as Safety Critical
+                UNDEFINED = 7               // UNDEFINED
+
+            };
+
+            inline std::optional<UPriority> UPriorityFromInt(int32_t value) {
+
+                switch (value) {
+                    case 0: {
+                        return UPriority::LOW;
+                    }
```

Comment:
Complexity and SRP violation. A data class like UPriority should map just data and leave to another class or operator any other conversions. Can we avoid this big switches using O(1) hash tables?